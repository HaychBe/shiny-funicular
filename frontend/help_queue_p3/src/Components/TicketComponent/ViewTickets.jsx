import React, { useState } from "react";
import { Button, Form, Modal } from "react-bootstrap";
import axios from 'axios';



const ViewTickets = ({
  id,
  title,
  traineeName,
  trainer,
  cohort,
  description,
  topic,
  completed
}) => {
  
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const [newCompleted, setCompleted] = useState(completed);

const completeTicket = (e) => {

    const ticketValues = {
        completed: newCompleted
      };


    axios
    .put("http://localhost:8080/ticket/completed/" + id, ticketValues, {
    headers: {
        'Access-Control-Allow-Origin' : '*'
        }
    })
    .then((response) => {
        console.log(response)
        setCompleted()
    })
    .catch((error) => {
        console.log(error);
    });

    handleClose();
}

if (!completed){

    return(
        <>
        <Button 
        variant="info" 
        onClick={handleShow}
        > View
        </Button>

        <Modal
            show={show}
            onHide={handleClose}
            dialogClassName="ticketModal"
            scrollable={true}
            >
            <Modal.Header closeButton>
                <Modal.Title>View Ticket Modal</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <Form onSubmit={completeTicket}>
                    <Form.Group>
                        <Form.Label>Trainee Name</Form.Label>
                        <Form.Control
                            readOnly value={traineeName}
                            />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Cohort</Form.Label>
                        <Form.Control
                            readOnly value={cohort}
                            >
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Topics</Form.Label>
                        <Form.Control
                            readOnly value={topic}
                            >

                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Trainer</Form.Label>
                        <Form.Control
                            readOnly value={trainer}
                            >
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Title</Form.Label>
                        <Form.Control
                            readOnly value={title}
                            />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Description</Form.Label>
                        <Form.Control
                            as="textarea"
                            rows={5}
                            readOnly value={description}
                            />
                    </Form.Group>
                </Form>

                <Button
                    onClick = {completeTicket}
                    type="submit"
                    variant="success"
                    size="lg"
                    block
                    >Complete Ticket
                
                </Button>
            </Modal.Body>
        </Modal>
    </>

)
} else {
    
    return(
        <>
        <Button 
        variant="info" 
        onClick={handleShow}
        > View
        </Button>

        <Modal
            show={show}
            onHide={handleClose}
            dialogClassName="ticketModal"
            scrollable={true}
            >
            <Modal.Header closeButton>
                <Modal.Title>View Ticket Modal</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <Form onSubmit={completeTicket}>
                    <Form.Group>
                        <Form.Label>Trainee Name</Form.Label>
                        <Form.Control
                            readOnly value={traineeName}
                            />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Cohort</Form.Label>
                        <Form.Control
                            readOnly value={cohort}
                            >
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Topics</Form.Label>
                        <Form.Control
                            readOnly value={topic}
                            >

                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Trainer</Form.Label>
                        <Form.Control
                            readOnly value={trainer}
                            >
                        </Form.Control>
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Title</Form.Label>
                        <Form.Control
                            readOnly value={title}
                            />
                    </Form.Group>

                    <Form.Group>
                        <Form.Label>Description</Form.Label>
                        <Form.Control
                            as="textarea"
                            rows={5}
                            readOnly value={description}
                            />
                    </Form.Group>
                </Form>
            </Modal.Body>
        </Modal>
    </>

)
}

};
export default ViewTickets;