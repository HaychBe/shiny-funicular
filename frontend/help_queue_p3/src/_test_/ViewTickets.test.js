import React from 'react';
import renderer from 'react-test-renderer';
import ViewTickets from  '../Components/TicketComponent/ViewTickets';





it('renders correctly', () => {
    const title = "Jest ";
    const topic = "Jest Test ";
    const traineeName = "Shamsi";
    const cohort = "cne";
    const trainer = "Reece";
    const description = "Jest Test description";
    const time = "0505";
    const tree = renderer.create(<ViewTickets title={title} topic={topic} traineeName={traineeName} cohort={cohort} trainer={trainer} description={description} time={time}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

