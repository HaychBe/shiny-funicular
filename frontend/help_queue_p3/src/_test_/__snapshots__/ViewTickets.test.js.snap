// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-info"
  disabled={false}
  onClick={[Function]}
  type="button"
>
   View
</button>
`;
